python/lsst/cp/pipe/ptc/cpPlotPtcTask.py

```diff
@@ -0,0 +1,1522 @@
+# This file is part of cp_pipe.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+__all__ = ["PlotPhotonTransferCurveConfig", "PlotPhotonTransferCurveTask"]
+
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+from matplotlib import gridspec
+from matplotlib.backends.backend_pdf import PdfPages
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+from lsst.cp.pipe._lookupStaticCalibration import lookupStaticCalibration
+
+from lsst.cp.pipe.utils import (
+    funcAstier,
+    funcPolynomial,
+    calculateWeightedReducedChi2,
+    getFitDataFromCovariances,
+)
+from matplotlib.ticker import MaxNLocator
+
+
+class PlotPhotonTransferCurveConnections(
+    pipeBase.PipelineTaskConnections, dimensions=("instrument", "exposure", "detector")
+):
+
+    dummyExposure = cT.Input(
+        name="raw",
+        doc="Dummy exposure to retreive PTC dataset.",
+        storageClass="Exposure",
+        dimensions=("instrument", "detector", "exposure"),
+        multiple=True,
+        deferLoad=True,
+    )
+    inputPtcDataset = cT.Input(
+        name="calib",
+        doc="Input PTC dataset.",
+        storageClass="PhotonTransferCurveDataset",
+        dimensions=("instrument", "detector"),
+        isCalibration=True,
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        doc="Camera associated with this data.",
+        storageClass="Camera",
+        dimensions=("instrument",),
+        isCalibration=True,
+        lookupFunction=lookupStaticCalibration,
+    )
+    # ptcFitType = "FULLCOVARIANCE" produces 12 plots
+    ptcPlot1 = cT.Output(
```

Comment:
Can you name these (or at least the output `dataset_type_name`) something more explicit, so that the on-disk file and the butler have something more human readable?  I'm not as worried about the internal usage, but being able to select a plot without having to look up if I want to look at `ptcPlot4` or `ptcPlot5` seems useful.

Reply:
I wanted to be able to plot all the individual files with a loop, something like this: 

from IPython.display import Image, display
detector = 55
for i in range(1,10):
    ref = butler.registry.findDataset(f'ptcPlot{i}', detector=detector)
    uri = butler.getURI(ref)
    display(Image(data=uri.read()))
    
I had also considered giving them more meaningful names. I can do that and will document the names in the ticket so the user can still do a loop like the above.

Reply:
Thanks, I had similar thoughts. Looks great now